frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx

```diff
@@ -25,19 +13,13 @@ import {
   compareStrings,
 } from '../../../../ui-components'
 import { DiffModal } from '../../components/DiffModal'
+import { useTemplateDetailsContext } from './PolicyTemplateDetailsPage'
 
-export function PolicyTemplateDetails(
-  props: Readonly<{
-    clusterName: string
-    template: any
-  }>
-) {
+export function PolicyTemplateDetails() {
   const { t } = useTranslation()
-  const { clusterName, template } = props
+  const { clusterName, template } = useTemplateDetailsContext()
   const isCertPolicy = template?.kind === 'CertificatePolicy'
   const [relatedObjects, setRelatedObjects] = useState<any>()
-  const [isExpanded, setIsExpanded] = useState<boolean>(isCertPolicy)
-  const [editorHeight, setEditorHeight] = useState<number>(250)
 
   const kind = template?.kind
```

Comment:
Could you please use the route param for the kind here instead? Otherwise, the card title says "undefined details" until the search query finishes.

Reply:
I think this should do it:
```diff
diff --git a/frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx b/frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx
index 3e45cec2..720c1c7a 100644
--- a/frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx
+++ b/frontend/src/routes/Governance/policies/policy-details/PolicyTemplateDetails.tsx
@@ -1,6 +1,7 @@
 /* Copyright Contributors to the Open Cluster Management project */
 import { Grid, GridItem, PageSection, Title } from '@patternfly/react-core'
 import { CheckCircleIcon, ExclamationCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons'
+import { useParams } from 'react-router-dom-v5-compat'
 import { useEffect, useMemo, useState } from 'react'
 import { useTranslation } from '../../../../lib/acm-i18next'
 import { NavigationPath } from '../../../../NavigationPath'
@@ -16,13 +17,14 @@ import { DiffModal } from '../../components/DiffModal'
 import { useTemplateDetailsContext } from './PolicyTemplateDetailsPage'
 
 export function PolicyTemplateDetails() {
+  const urlParams = useParams()
+  const kind = urlParams.kind ?? ''
+
   const { t } = useTranslation()
   const { clusterName, template } = useTemplateDetailsContext()
-  const isCertPolicy = template?.kind === 'CertificatePolicy'
+  const isCertPolicy = kind === 'CertificatePolicy'
   const [relatedObjects, setRelatedObjects] = useState<any>()
 
-  const kind = template?.kind
-
   useEffect(() => {
     if (template?.status?.relatedObjects?.length) {
       const relObjs = template.status.relatedObjects.map((obj: any) => {
@@ -75,7 +77,7 @@ export function PolicyTemplateDetails() {
     },
     {
       key: t('Kind'),
-      value: template?.kind ?? '-',
+      value: kind ?? '-',
     },
     {
       key: t('API groups'),
```

Reply:
Amazing catch! Alien!!!!!